cmd/devp2p/internal/ethtest/snap.go

```diff
@@ -470,13 +469,13 @@ func (s *Suite) TestSnapGetByteCodes(t *utesting.T) {
 			desc:      `Here we request state roots as code hashes. The server should deliver an empty response with no items.`,
 			nBytes:    10000,
 			hashes:    []common.Hash{genesisRoot, headRoot},
-			expHashes: 1, // 32-byte keys are detected as code, even if not code (like genesis hash), in legacy lookups.
+			expHashes: 0,
 		},
 		{
 			desc:      `Here we request the genesis state root (which is not an existing code hash) two times. The server should deliver an empty response with no items.`,
 			nBytes:    10000,
 			hashes:    []common.Hash{genesisRoot, genesisRoot},
-			expHashes: 2, // 32-byte keys are detected as code, even if not code (like genesis hash), in legacy lookups.
+			expHashes: 0,
```

Comment:
This PR removes this explicit OP modification. I'm assuming this is due to a difference in the underlying DB structure -- but would appreciate a sanity check from someone who has more familiarity with the snapsync code.

Reply:
Snap-sync has contract-code retrieval, which used to be namespaced differently in geth. We still support syncing from/to the old database contents, and thus had to modify the tests to check what is valid and what is not. I will check if we need to backport this DB compatibility thing.

Reply:
I checked, and the `ServiceGetByteCodesQuery` in `eth/protocols/snap/handler.go` still calls `ContractCode` like in our op-stack diff, which will then do the new DB lookup + fall back to legacy DB lookup when no data is found.
So our diff is ok, but the tests changing is odd.

Reply:
It looks like the test context changed, and so now there is no code to retrieve, and the tests will pass without modifying expectations.